src/components/Container/index.tsx

```diff
@@ -0,0 +1,42 @@
+/* eslint-disable react/jsx-props-no-spreading */
+
+import React, {FC, useMemo} from 'react';
+import clsx from 'clsx';
+
+import './Container.scss';
+
+type ContainerElement = 'div' | 'header' | 'main';
```

Comment:
should we have `section` too? 🤔

Reply:
yeah we can have all the elements. But i just added the ones that were already used & in need of this component. Feel free to add more to this list in the future whenever you feel the need